src/java/org/apache/cassandra/streaming/StreamingState.java

```diff
@@ -70,6 +72,15 @@ public class StreamingState implements StreamEventHandler
     // API for state changes
     public final Phase phase = new Phase();
 
+    @Override
+    public long unsharedHeapSize()
+    {
+        long costOfPeers = peers().size() * (InetAddressAndPort.IPV6_SIZE + 48); // 48 represents the datastructure cost computed by the JOL
+        long costOfCompleteMessage = ObjectSizes.sizeOf(completeMessage());
+        long weight = costOfPeers + costOfCompleteMessage + EMPTY;;
```

Comment:
don't store `weight` into a variable, just return it; there is an IntelliJ warning showing this feedback as well

Reply:
you have an extra `;`